File: android-java/JavaApp1/app/src/main/java/com/example/javaapp1/MainActivity.java
package com.example.javaapp1;

import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.GravityCompat;
import androidx.navigation.Navigation;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.NavigationUI;

import android.app.Activity;
import android.os.Bundle;
import android.view.TextureView;
import android.view.View;
import android.widget.TextView;

import com.example.javaapp1.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        //drawer menüsünü bağlama satırları
        NavHostFragment navHostFragment  =
                (NavHostFragment) getSupportFragmentManager().findFragmentById(R.id.nav_host_fragment);
        NavigationUI.setupWithNavController(binding.navigationView,navHostFragment.getNavController());
        binding.toolbar.setTitle("Başlık");
        ActionBarDrawerToggle toggle=
                new ActionBarDrawerToggle(this,binding.drawer,binding.toolbar,0,0);
        binding.drawer.addDrawerListener(toggle);
        toggle.syncState();

        // drawedaki başlığı bu satırlarla yerleştirdik
        View baslik = binding.navigationView.inflateHeaderView(R.layout.drawer_baslik);
        TextView textViewBaslik = baslik.findViewById(R.id.textViewbaslik);
        textViewBaslik.setText("Merhaba"); // setText ile başka başlık koyduk

    }


}

File: android-kotlin/BasicCalculator/app/src/main/java/com/example/basiccalculator/MainActivity.kt
package com.example.basiccalculator

import android.os.Bundle
import android.view.View
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import com.example.basiccalculator.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding : ActivityMainBinding
    private  var firstNumber: Double = 0.0
    private  var secondNumber: Double= 0.0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityMainBinding.inflate(layoutInflater)

        val view = binding.root
        setContentView(view)









    }
    fun isValidNumber(): Boolean {
        if (binding.number1Text.text.toString().toDoubleOrNull() == null || binding.number2Text.text.toString().toDoubleOrNull() == null) {
            binding.resultText.setText("Please enter a valid number")
            return false
        }
        else{
            firstNumber = binding.number1Text.text.toString().toDouble()
            secondNumber = binding.number2Text.text.toString().toDouble()
            return true

        }
    }
    fun onSumClick(view: View)  {
        if (isValidNumber()) {
            val result = firstNumber + secondNumber
            binding.resultText.text = "Result: $result"
            binding.number1Text.text.clear()
            binding.number2Text.text.clear()
        }



    }fun onSubstractClick(view: View)  {
        if(isValidNumber()){
            val result = firstNumber - secondNumber
            binding.resultText.text = "Result: $result"
            binding.number1Text.text.clear()
            binding.number2Text.text.clear()
        }


    }
    fun onMultiplyClick(view: View)  {
        if(isValidNumber()){
            val result = firstNumber * secondNumber
            binding.resultText.text = "Result: $result"
            binding.number1Text.text.clear()
            binding.number2Text.text.clear()
        }


    }
    fun onDivideClick(view: View)  {
        if (isValidNumber()){
            val result = firstNumber / secondNumber
            binding.resultText.text = "Result: $result"
            binding.number1Text.text.clear()
            binding.number2Text.text.clear()
        }


    }

}